Sources/OpenAI/OpenAI.swift

```diff
@@ -171,7 +175,7 @@ extension OpenAI {
         var components = URLComponents()
         components.scheme = "https"
         components.host = configuration.host
-        components.path = path
+        components.path = (configuration.basePath ?? "") + path
```

Comment:
What do you think of improving reliability here?
We can use `appendingPathComponent` or a similar function to handle cases when "/path", "/path/" and "path/" are supplied.
Extracting this to the function and several unit tests would be a great idea. 
What do you think?

Reply:
Great suggestions, code refactored.